tests/src/test_layout_cache.cpp

```
@@ -0,0 +1,64 @@
+#include "mock_context.h"
+#include "moth_ui/layout/layout_cache.h"
+#include "moth_ui/layout/layout.h"
+#include <catch2/catch_all.hpp>
+#include <filesystem>
+#include <fstream>
+
+using namespace moth_ui;
+
+namespace {
+    struct TempDir {
```

Clang (14.0.6)

[error] class 'TempDir' defines a non-default destructor but does not define a copy constructor, a copy assignment operator, a move constructor or a move assignment operator

(cppcoreguidelines-special-member-functions,-warnings-as-errors)